File: Baas/justep/Push.java
package justep;

import java.io.InputStream;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.UUID;

import org.dom4j.Document;
import org.dom4j.Element;
import org.dom4j.io.SAXReader;

import cn.jpush.api.JPushClient;
import cn.jpush.api.common.ClientConfig;
import cn.jpush.api.common.connection.HttpProxy;
import cn.jpush.api.common.resp.APIConnectionException;
import cn.jpush.api.common.resp.APIRequestException;
import cn.jpush.api.push.model.Platform;
import cn.jpush.api.push.model.PushPayload;
import cn.jpush.api.push.model.audience.Audience;
import cn.jpush.api.push.model.notification.Notification;
import cn.jpush.api.schedule.ScheduleResult;

import com.alibaba.fastjson.JSONObject;
import com.justep.baas.action.ActionContext;

public class Push {
    
    private static String appKey = "";
    private static String masterSecret = "";
    private static Boolean apnsProduction = false;
    private static JPushClient jpushClient;
    private static Boolean useProxy = false;
    
    static{
        InputStream configFile = Push.class.getResourceAsStream("jpush.config.xml");
        try{
            SAXReader reader = new SAXReader();
            Document doc = reader.read(configFile);
            Element config = doc.getRootElement();
            appKey = config.elementTextTrim("appKey");
            masterSecret = config.elementTextTrim("masterSecret");
            apnsProduction = (config.elementTextTrim("apnsProduction").equals("true"))?true:false;
        }catch(Exception e){
            e.printStackTrace();
        }
    }
    
    public static JSONObject push(JSONObject params, ActionContext context){
        String registrationId = params.getString("registrationId");
        try {
            sendPushMessage(registrationId, appKey, masterSecret);
        } catch (APIConnectionException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        } catch (APIRequestException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
        return null;
    }
    
    
    public static ScheduleResult sendPushMessage(String registrationId, String key, String secret) throws APIConnectionException, APIRequestException{
        ClientConfig config = ClientConfig.getInstance();
        HttpProxy proxy = null;
        if (useProxy) {
            proxy = new HttpProxy("http-proxy.system", 3128);
        }
        jpushClient = new JPushClient(secret, key, 3, proxy, config);
        PushPayload payload = PushPayload.newBuilder()
        .setPlatform(Platform.all())
        .setAudience(Audience.registrationId(registrationId))
        .setNotification(Notification.alert("来自极光推送的通知~"))
        .build();
        payload.resetOptionsTimeToLive(86400);
        payload.resetOptionsApnsProduction(apnsProduction);
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        Calendar nowTime = Calendar.getInstance();
        //修改为60秒后推送，10秒推送容易超时导致推送失败。
        nowTime.add(Calendar.SECOND, 60);
        String scheduleTime =sdf.format(nowTime.getTime());
        ScheduleResult result = jpushClient.createSingleSchedule(UUID.randomUUID().toString().replaceAll("-", ""), scheduleTime, payload);
        return result;
    }
}

File: Baas.net/jpush/Push.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Web;
using System.Xml;
using cn.jpush.api;
using cn.jpush.api.common;
using cn.jpush.api.common.resp;
using cn.jpush.api.push;
using cn.jpush.api.push.mode;
using Newtonsoft.Json.Linq;

namespace com.justep.Sample.JPush
{
    using JSONObject = JObject;
    public class Push
    {
        private static string appKey = "";
        private static string masterSecret = "";
        private static bool apnsProduction = false;
        private static JPushClient jpushClient;

        static Push()
        {
            var configFile = Assembly.GetEntryAssembly().Location + "jpush/jpush.config.xml";
            
            try
            {
                XmlDocument doc = new XmlDocument();
                doc.Load(configFile);
                var config = doc.DocumentElement;
                if (config != null)
                {
                    appKey = config.GetElementsByTagName("appKey").ToString();
                    masterSecret = config.GetElementsByTagName("masterSecret").ToString();
                    apnsProduction = (config.GetElementsByTagName("apnsProduction").ToString() == "true");
                }
            }
            catch (Exception e)
            {
                throw new SystemException(e.Message, e);
            }

        }

        public static JSONObject push(HttpContext context)
        {
            String registrationId = context.Request.Params["registrationId"];
            try
            {
                sendPushMessage(registrationId, appKey, masterSecret);
            }
            catch (APIConnectionException e)
            {
                throw new SystemException(e.Message, e);
            }
            catch (APIRequestException e)
            {
                throw new SystemException(e.Message, e);
            }
            return null;
        }

        public static MessageResult sendPushMessage(string registrationId, string key, string secret)
        {

            //ClientConfig config = ClientConfig.getInstance();
            //jpushClient = new JPushClient(secret, key, 3, null, config);
            jpushClient = new JPushClient(key, secret);
            PushPayload payload = new PushPayload();
            payload.platform = Platform.all();
            payload.audience = Audience.s_registrationId(registrationId);
            payload.notification = new Notification().setAlert("亲爱的x5外卖用户，您的订单已经配货完毕，正在运送中..");

            payload.ResetOptionsTimeToLive(86400);
            payload.ResetOptionsApnsProduction(apnsProduction);
            string schedleTime = DateTime.Today.ToString("yyyy-MM-dd HH:mm:ss");
            MessageResult result = jpushClient.SendPush(payload);
            return result;
        }
    }
}
